controller.py

```diff
@@ -38,11 +41,30 @@ def rebuild_from_logs(self):
         logger.info(f"Starting to read previous journal files: {journal_folder}")
 
         logs = [f for f in os.listdir(journal_folder) if ".log" in f]  # get logs only
-        logs.sort(reverse=True)
-        logs = logs[0:7]  # Get the last 7 days
-        logs.sort()
-        files_with_issues = []
+
+        # Calculate a week ago
+        now = datetime.datetime.now()
+        week_ago = now - timedelta(days=7)
+
+        # Then build an array with the wanted logs
+        wanted_logs = []
+        found_first_log = False
         for log in logs:
+            if not found_first_log:
+              match = re.search(r"^Journal\.(\d\d)(\d\d)(\d\d)(\d\d)", log)
```

Comment:
I'd compile this outside the `for` loop, then use the compiled matcher object inside of it. 

Something like this, I think:

```py
matcher = re.compile(r"^Journal\.(\d\d)(\d\d)(\d\d)(\d\d)")
for log in logs:
  match = matcher.search(log)
```